cucumber/authenticators_jwt/features/support/jwt_jwks_helper.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+# Utility methods for JWT and JWKs manipulation
+
+require 'openssl'
+require 'jwt'
+
+module JWTJWKsHelper
+
+  module Algorithms
+    RS256 = "RS256".freeze
+  end
+
+  JWKS_ROOT_PATH = "/var/jwks".freeze
+  JWKS_BASE_URI = "http://jwks".freeze
+  BITS_2048 = 2048
+  HOUR_IN_SECONDS = 3600
+
+  def init_jwks_file(file_name)
+    jwks = { keys: [jwk.export] }
+    File.write("#{JWKS_ROOT_PATH}/#{file_name}", JSON.pretty_generate(jwks))
+  end
+
+  def issue_jwt_token(token_body, algorithm = Algorithms::RS256)
+    @jwt_token ||= JWT.encode(
+      token_body,
+      rsa_key,
+      algorithm,
+      { kid: jwk.kid }
+    )
+  end
```

Comment:
a few points here:
1. Memoizing functions should have the same name as the object they are memoizing
2. A memoizing function can't get inputs. this is because the second call to the function will not do anything with the input as the method is already initialized.

it seems that you want to remove the memoization here.

Reply:
The rule of "memorizing" looks me in-applicable for this concrete use-case.
When [this](https://github.com/cyberark/conjur/blob/8e4f58a9b313cda89722866e126e71801886486f/cucumber/authenticators_jwt/features/step_definitions/authn_jwt_steps.rb#L9) step is using the token it can not provide any valuable input to the `issue` function.
I don't want to be in situation where the step is looking like this:
```
When(/I authenticate via authn-jwt with the JWT token/) do
  authenticate_jwt_token(jwt_token(nil, nil))
end
```

Reply:
We've talked. As a result I've removed `||` sign from assignment this keeps the class state-full but makes `@jwt_token` rewritable.